modules/tests/src/test/scala/testkit.scala

```diff
@@ -78,26 +75,29 @@ def request[F[_]: RefConstructor: MonadThrow, T <: requests.LSPRequest](
     builder
       .build(communicate)
       .find(_.method == req.requestMethod)
-      .collectFirst {
-        case Endpoint.RequestResponseEndpoint(_, run, inc, erc, outc) =>
-          val encoded = Payload(upickle.default.write(in).getBytes())
-          for
-            lifted <- F.fromEither(inc.decode(Some(encoded)))
-            msg = RequestMessage(
-              method = req.requestMethod,
-              callId = id,
-              params = Some(encoded)
-            )
-            res <- run(msg, lifted).flatMap {
-              case Left(err) => F.raiseError(erc.encode(err))
-              case Right(res) =>
-                F.catchNonFatal {
-                  upickle.default.read[req.Out](writeToArray(outc.encode(res)))
-                }
-            }
-          yield res -> communicate
-          end for
+      .collectFirst { case _ =>
+        F.pure(null.asInstanceOf[req.Out] -> communicate)
       }
+      // .collectFirst {
```

Comment:
if we decided to keep the LSPTest in the current form (with the use of fs2Channel) then most of this file can be deleted.